exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/_metric_exporter/__init__.py

```diff
@@ -114,9 +122,43 @@ def _translate_data(
             instrumentation_library_metrics = instrumentation_library_map.get(
                 metric.instrumentation_info
             )
+            
+            # translate all data points for that metric
+            self._collector_kwargs["name"] = metric.name
+            self._collector_kwargs["description"] = metric.description
+            self._collector_kwargs["unit"] = metric.unit
+            if isinstance(metric.point, Gauge):
+                # TODO: implement
+                self._collector_kwargs["gauge"] = OTLPGauge(
+                    data_points=[],
+                )
+            elif isinstance(metric.point, Histogram):
+                # TODO: implement
+                self._collector_kwargs["histogram"] = OTLPHistogram(
+                    data_points=[],
+                )
+            elif isinstance(metric.point, Sum):
+                args = {
+                    "attributes":self._translate_attributes(metric.attributes),
+                    "start_time_unix_nano":metric.point.start_time_unix_nano,
+                    "time_unix_nano":metric.point.time_unix_nano,
+                }
+                if isinstance( metric.point.value, int):
+                    args["as_int"] = metric.point.value
+                else:
+                    args["as_double"] = metric.point.value
+                pt = OTLPNumberDataPoint(**args)
```

Comment:
High level question, is there a reason to use bare strings and **kwargs here over the generated proto types? We also have generated type stubs for all the protos so you should get nice autocomplete (and ideally type checking). So for example:

```py
pt = OTLPNumberDataPoint(
    attributes=self._translate_attributes(metric.attributes),
    start_time_unix_nano=metric.point.start_time_unix_nano,
    time_unix_nano=metric.point.time_unix_nano,
)
if isinstance(metric.point.value, int):
    pt.as_int = metric.point.value
else:
    pt.as_double = metric.point.value
```

same for Sum and Metric points

Reply:
No particular reason, it's the same approach that was used in the log/trace exporters, so i followed that pattern.

Reply:
Changed, I agree this is a bit better to use this, maybe we can change it for the other exporters as well.